orderer/consensus/etcdraft/chain.go

```diff
@@ -911,6 +921,41 @@ func (c *Chain) ordered(msg *orderer.SubmitRequest) (batches [][]*common.Envelop
 			}
 		}
 
+		if !c.leaderTransferFailed {
```

Comment:
if the transaction has nothing to do with removal of the leader, you still set the transferFailed to false, why?

Reply:
now the `else block` resets only if leaderTransferFailed is set in previous iteration.